### ISSUE TITLE
Using a different docker volume after running dokku couchdb:upgrade <app> ?

### ISSUE THREAD

**Issue by knubie** [dokku/dokku-couchdb#51]
I'm sure this probably isn't the right place to post this, but I'm a kind of desperate for help at this point.

I tried to upgrade my dokku couchdb instance from 2.3.1 to 3.2.1 using `dokku couchdb:upgrade`. I'm not sure what happened, but the couchdb service became unreachable. I tried checking the logs with `dokku couchdb:logs` which said it couldn't find the `_users` directory. At this point I tried to downgrade back to 2.3.1 with `dokku couchdb:upgrade --image-version 2.3.1`, but now service runs, but all of the databases are gone.

At this point I tried looking for the data and was able to find it on a docker volume, but it seems the couchdb service is now using a different volume and I'm not sure how to go back to using the original volume, or how to restore. Any help would be much appreciated.

**Comment by josegonzalez** [comment]
This plugin doesn't use docker volumes, it writes files to disk directly. You can check the path on disk via `dokku couchdb:info $SERVICE`. Did you create a backup before upgrading? If so, you can create a new service and import that backup.

**Comment by knubie** [comment]
Thanks @josegonzalez for the fast reply. I do wonder why I was able to find the data in a docker volume in that case...  It was in a volume like `/var/lib/docker/volumes/829f13aea3499d512fc4f8f71rb77e1dc162148443a1d779047bf475e7f98bbf/_data`.

At this point I've copied the contents of that volume back to `/var/lib/dokku/services/couchdb/lollipop/data`. Still not sure what happened. Again I appreciate your help.

**Comment by josegonzalez** [comment]
Not sure either, but as you see [here](https://github.com/dokku/dokku-couchdb/blob/master/functions#L87), we use a directory on disk to back the data volume.

Glad to see you're back up and running (always backup and test your backups!). Feel free to donate to the project if you find it useful, I personally use the money for either cat food or buying myself drinks at a bar, so you know it'll go to a good cause :)

Closing since hopefully there is nothing to do here now.

**Comment by knubie** [comment]
Thanks Jose,

I'm still going through the wreckage to try and piece together what actually happened but I'm totally stumped. You're right that the couchdb dokku plugin doesn't create docker volumes, so I have no idea why there would be a volume containing the `data` directory. The only thing I can think of is that I created the service using an older version of the couchdb plugin, and maybe that was using volumes?

I tried running the same upgrade sequence on a backup server created from a snapshot, and everything went as expected. If I come up with anything else, I'll post an update here.

And I will definitely be sponsoring the project now. I definitely appreciate the work you're doing. Cats need to eat too!
